cluster/ClusterDatabase.java

```diff
@@ -41,11 +41,13 @@
     private static final Logger LOG = LoggerFactory.getLogger(ClusterDatabase.class);
     private final String name;
     private final Map<String, CoreDatabase> databases;
+    private ClusterClient client;
     private final ClusterDatabaseManager databaseMgr;
     private final Set<Replica> replicas;
 
-    private ClusterDatabase(String database, ClusterDatabaseManager clusterDatabaseMgr) {
+    private ClusterDatabase(String database, ClusterClient client, ClusterDatabaseManager clusterDatabaseMgr) {
```

Comment:
If we're passing in the whole Client, might as well grab `databases` from it:
```suggestion
    private ClusterDatabase(String database, ClusterClient client) {
```

```java
this.databaseMgr = client.databases();
```